File: ticket-backend/src/main/java/aics/domain/fuzzy/models/FuzzyVariableDuration.java
package aics.domain.fuzzy.models;

import aics.domain.fuzzy.constants.FuzzyVariableDurationFields;
import aics.domain.fuzzy.constants.FuzzyVariableYearFields;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.HashMap;

@Data
@Accessors(chain = true)
@AllArgsConstructor
public class FuzzyVariableDuration implements Serializable, FuzzyVariableI {
    private FuzzyVariableDistributionPart varSmall;
    private FuzzyVariableDistributionPart varAverage;
    private FuzzyVariableDistributionPart varBig;
    private FuzzyVariableDistributionPart varHuge;

    @Override
    public HashMap<String, FuzzyVariableDistributionPart> getFuzzyVariableMap() {
        HashMap<String, FuzzyVariableDistributionPart> fuzzyVariableMap = new HashMap<>();
        fuzzyVariableMap.put(FuzzyVariableDurationFields.SMALL.name(), this.varSmall);
        fuzzyVariableMap.put(FuzzyVariableDurationFields.AVERAGE.name(), this.varAverage);
        fuzzyVariableMap.put(FuzzyVariableDurationFields.BIG.name(), this.varBig);
        fuzzyVariableMap.put(FuzzyVariableDurationFields.HUGE.name(), this.varHuge);
        return fuzzyVariableMap;
    }
}

File: ticket-frontend/src/modules/fuzzy/fuzzy-variable-duration.ts
import { FuzzyVariableDistributionPart, FuzzyVariableDistributionPartUtils, FuzzyVariableI } from './fuzzy-variable-distribution';

export class FuzzyVariableDuration implements FuzzyVariableI {
    varSmall: FuzzyVariableDistributionPart;
    varAverage: FuzzyVariableDistributionPart;
    varBig: FuzzyVariableDistributionPart;
    varHuge: FuzzyVariableDistributionPart;

    constructor(obj: {
        varSmall: FuzzyVariableDistributionPart,
        varAverage: FuzzyVariableDistributionPart,
        varBig: FuzzyVariableDistributionPart,
        varHuge: FuzzyVariableDistributionPart
    }) {
        this.varSmall = obj.varSmall;
        this.varAverage = obj.varAverage;
        this.varBig = obj.varBig;
        this.varHuge = obj.varHuge;
    }

    static fromObj(obj: any): FuzzyVariableDuration {
        return new FuzzyVariableDuration({
            varSmall: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varSmall),
            varAverage: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varAverage),
            varBig: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varBig),
            varHuge: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varHuge)
        })
    }

    getFuzzyVariableMap(): Record<string, FuzzyVariableDistributionPart> {
        return {
            [FuzzyVariableDurationFields.SMALL]: this.varSmall,
            [FuzzyVariableDurationFields.AVERAGE]: this.varAverage,
            [FuzzyVariableDurationFields.BIG]: this.varBig,
            [FuzzyVariableDurationFields.HUGE]: this.varHuge,
        }
    }

    getFuzzyVariableColorsMap(): Record<string, string> {
        return {
            [FuzzyVariableDurationFields.SMALL]: 'red',
            [FuzzyVariableDurationFields.AVERAGE]: 'orange',
            [FuzzyVariableDurationFields.BIG]: 'yellow',
            [FuzzyVariableDurationFields.HUGE]: 'green',
        }
    }

}

export enum FuzzyVariableDurationFields {
    SMALL = 'SMALL',
    AVERAGE = 'AVERAGE',
    BIG = 'BIG',
    HUGE = 'HUGE',
}